spyica/core.py

```diff
@@ -262,11 +260,11 @@ def ica_alg(recording, fs, clustering='mog', n_comp='all',
             n_chunks=0, chunk_size=0, spike_thresh=5, dtype='int16',
             keep_all_clusters=False, verbose=True):
     if n_comp == 'all':
-        n_comp = recording.shape[0]
+        n_comp = recording.get_num_channels()
     if verbose:
         print('Applying FastICA')
         t_init = time.time()
-    traces = recording
+    traces = recording.get_traces().astype(dtype)
```

Comment:
```suggestion
    traces = recording.get_traces().astype(dtype).T
```
Shouldn't this be transposed?